djconnectwise/models.py

```diff
@@ -1552,9 +1570,25 @@ def update_cw(self, **kwargs):
             api_private_key=kwargs.get('api_private_key')
         )
 
-        return api_client.update_ticket(
-            self.id, self.closed_flag, self.priority, self.status
-        )
+        # Get any fields changed since last save when called from a
+        # TicketTracker instance.
+        tracker = getattr(self, 'tracker', None)
+        if tracker:
+            changed_fields = tracker.changed()
+
+        else:
+            changed_fields = kwargs.get('changed_fields')
```

Comment:
This one is a question: Could you give me a short explanation of these two cases? Thanks Cameron.

Reply:
The two cases are:
1. `update_cw` is called from a TicketTracker instance. This is the case that will be used when the ticket edit dialog is added. We'll make sure that the calling code in the dialog uses TicketTracker so it has access to any changed fields.
1. `update_cw` is called from card_detail.py for status and priority updates. I couldn't use TicketTracker here because the request originates from a ConnectWiseServiceCard object.

Reply:
>     called from card_detail.py for status and priority updates

Do we need this case? This status/priority features are going away in #1796- and anyway needing it in django-connectwise suggests a less-than-ideal design here. Ideally we should be designing django-connectwise to be able to be used in any project- and things in Kanban that drive design in django-connectwise indicate a problem. (Not always possible, but it's the ideal.)

Reply:
I believe we'll still need status changes through here for change status via columns. But yes, I realize priority won't need to be changed from here after #1796. 

It's a good point though. Maybe I could improve it by making the code that calls `update_cw` responsible for providing any changed fields? Either from the tracker.changed() or on column status changes from card_detail. Then `update_cw` just has to take the changed data, format it and send to CW.